### PR TITLE
Refactor verifySignature code

### LINKED ISSUE #286: web3 refactoring code
in verifySignature, create a public key object instead of that code:
```
 // setup the public key.
    const versionAndPublicKeyBytes: Uint8Array =
      getBytesPublicKey(signerPubKey);
    const publicKeyBytes = versionAndPublicKeyBytes.slice(1);
```
creating a new constructor to create a public key object from the string will be needed

### PR DESCRIPTION
**Pull request by julienbrs**
closes #286 

Implements static method to create a public key from a base58encoded string and find the version number from it.
Refactor some code thanks to that

### CODE REVIEW

File: src/utils/keyAndAdresses.ts [hunk 1]
```diff
@@ -69,13 +70,25 @@ export class PublicKey {
       PUBLIC_KEY_PREFIX +
       base58Encode(Buffer.concat([versionBuffer, Buffer.from(this.bytes)]));
   }
+
+  /* Static method to create a public key from a base58 encoded string, and decoding the version number */
```
Comment (Ben-Rey) [inline src/utils/keyAndAdresses.ts:74]:
```suggestion
  // Create a new PublicKy object from a base58 encoded string
```

File: src/utils/keyAndAddresses.ts [hunk 2]
```diff
@@ -69,15 +70,27 @@ export class PublicKey {
       PUBLIC_KEY_PREFIX +
       base58Encode(Buffer.concat([versionBuffer, Buffer.from(this.bytes)]));
   }
+
+  // Create a new PublicKy object from a base58 encoded string
```
Comment (Ben-Rey) [inline src/utils/keyAndAddresses.ts:74]:
```suggestion
  // Create a new PublicKey object from a base58 encoded string
```

File: src/utils/keyAndAddresses.ts [hunk 3]
```diff
@@ -69,15 +70,27 @@ export class PublicKey {
...
 
 /**
  * An address.
  *
  * @remarks the address object is created from a public key and got the same version as the public key.
  *
+ * @remarks String representation is A + U/S + base58Check(version_bytes + hashBlake3(version_bytes + public_key_bytes))
  * @remarks the address bytes representation is `version + hashBlake3(version + publicKey)`.
- * @remarks publicKey bytes is not an attribute of the address object because it is not needed.
+ * @remarks bytes is not an attribute of the address object because it is not needed.
```
Comment (Ben-Rey) [inline src/utils/keyAndAddresses.ts:93]: It's best to have one remarks with multiple points:

```typescript
@remarks
    - first remark
    - second
    ...
```

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

**Review by Ben-Rey — APPROVED** [review]

File: src/utils/keyAndAddresses.ts [hunk 4]
```diff
@@ -69,15 +71,28 @@ export class PublicKey {
       PUBLIC_KEY_PREFIX +
       base58Encode(Buffer.concat([versionBuffer, Buffer.from(this.bytes)]));
   }
+
+  // Create a new PublicKey object from a base58 encoded string
+  static fromString(publicKeyBase58Encoded: string): PublicKey {
+    const versionAndKeyBytes = getBytesPublicKey(publicKeyBase58Encoded);
```
Comment (gregLibert) [inline src/utils/keyAndAddresses.ts:77]:
```suggestion
  static fromString(base58Encoded: string): PublicKey {
    const versionAndKeyBytes = getBytesPublicKey(base58Encoded);
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**